src/clr-angular/utils/host-wrapping/host-wrapper.ts

```diff
@@ -0,0 +1,52 @@
+/**
+ * Copyright (c) 2016-2017 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import {
+    ComponentFactory,
+    ComponentFactoryResolver,
+    ElementRef,
+    InjectionToken,
+    Injector,
+    Type,
+    ViewContainerRef
+} from "@angular/core";
+
+import {DynamicWrapper} from "./dynamic-wrapper";
+import {EmptyAnchor} from "./empty-anchor";
+
+export class HostWrapper<W extends DynamicWrapper> implements Injector {
+    // TODO: injector might be optional, we can get it from the ViewContainerRef
+    constructor(containerType: Type<W>, vcr: ViewContainerRef) {
+        this.injector = vcr.injector;
+        // If the host is already wrapped, we don't do anything
+        if (!this.injector.get(containerType, null)) {
+            const cfr = this.injector.get(ComponentFactoryResolver);
+            const el = this.injector.get(ElementRef);
+
+            // We need a new anchor, since we're projecting the current one.
+            vcr.createComponent(cfr.resolveComponentFactory(EmptyAnchor));
+            const factory: ComponentFactory<W> = cfr.resolveComponentFactory(containerType);
+            // We're assuming only one projection slot, but in more complex cases we might want to provide
+            // a different array of projected elements.
+            const containerRef = vcr.createComponent(factory, undefined, undefined, [[el.nativeElement]]);
+            // We can now remove the useless anchor
+            vcr.remove(0);
+
+            // We note that the container was dynamically created
+            // (no idea why we have to force cast, typescript is weird since I updated)
+            containerRef.instance._dynamic = true;
```

Comment:
I don't understand this comment. What do you mean that you had to force cast this? What would you have rather done if TS wasn't acting weird?

Reply:
What's the point of noting that the wrapper was dynamically created? We are not using that property anywhere. Am I missing something?

Reply:
The `_dynamic` property is meant to be consumed in the template of a container implementation so it can add a label if it is missing. Its a quirk specific to radio and checkboxes at least.

Reply:
Comment from a previous version, I was able to drop the force cast but forgot to remove the comment. Thanks for catching that.

And as @gnomeontherun pointed out, it provides the container with a way to perform specific work when generated dynamically. For radios and checkboxes, it can create an empty label. For email inputs, it could add a default error if the value isn't an email. There could be more cases, and it didn't cost much to put it there.